thirdweb/modules/bundle.py

```diff
@@ -91,20 +96,50 @@ def create_and_mint_batch(self, meta_with_supply: List[CreateBundleArg]) -> List
         token_ids = result[0]['args']['tokenIds']
         return [self.get(i) for i in token_ids]
 
-    def create_with_erc20(self, token_contract: str, token_amount: int, arg: CreateBundleArg):
-        uri = self.get_storage().upload(
-            arg.metadata, self.address, self.get_signer_address())
+    def create_with_token(self, token_contract: str, token_amount: int, metadata: dict = None):
+        uri = self.upload_metadata(metadata)
+        if token_contract is not None and token_contract != ZeroAddress:
+            erc20 = ERC20(self.get_client(), token_contract)
+            allowance = erc20.allowance.call(self.get_signer_address(), self.address)
+            if allowance < token_amount:
+                tx = erc20.increase_allowance.build_transaction(self.address,
+                                                                token_amount,
+                                                                self.get_transact_opts()) 
+                self.execute_tx(tx)
+
         self.execute_tx(self.__abi_module.wrap_erc20.build_transaction(
-            token_contract, token_amount, arg.supply, uri, self.get_transact_opts()
+            token_contract, token_amount, token_amount, uri, self.get_transact_opts()
         ))
-
-    def create_with_erc721(self, token_contract: str, token_id: int, metadata):
-        uri = self.get_storage().upload(
-            metadata.metadata, self.address, self.get_signer_address())
+        
+    def create_with_nft(self, token_contract: str, token_id: int, metadata):
```

Comment:
Thanks for filling in these methods, but lets try to keep our PRs clean. So for this PR, it should be *only* the upload support, but I'll let this one through because its been open for so long. We don't want massive PRs that contain too many changes, especially when they don't have great test coverage.